perf/evd/classic/evd_classic_perf.cpp

```diff
@@ -9,6 +9,17 @@
 using EVDEpochType = decltype(&evd_classic);
 using EVDTolType = decltype(&evd_classic_tol);
 
+double base_cost_evd(int n, int n_iter) {
```

Comment:
Shouldn't the cost contain at least one `n^3` term? This function should return the exact cost of the whole algorithm, not per element of the matrix.

Reply:
In theory, yes. When we were using matrix multiplication, it was containing `n^3` term. But if you observe the optimized algorithm, it's runs (when floating point operations are considered) on O(n) for each value (largest off-diagonal element corresponding to each epoch), i.e. only 2 rows and 2 columns of the input matrix and 2 columns of the eigenvector matrix are getting updated. Since it runs for the number of epochs specified, the whole algorithm becomes O(n * n_iter) as n_iter is also a parameter. But to be more specific, the algorithm runs for O(n^2) times in each epoch if you consider computing the largest off-diagonal entry, which involves only comparisons and not any other operation. This leaves me with another question - should we consider comparison as a separate floating point operation?

Reply:
Moreover, if you observe the latest update for cyclic evd, it contains `n^3` term since for each and every epoch, we perform Jacobi rotation and update for all elements in the upper triangular matrix.

Reply:
A kind of contrast that can be observed between both approaches is that the value of n_iter plays an important role. For classic evd, it's initialized as 100 (might require more for very large size matrices, we need to investigate it) and for cyclic evd, it's 20. That also can create an impact because classic evd requires more epochs to converge.

Reply:
I just checked `evd_classic` implementation. What do you mean by `O(n)`? There are two matrix multiplications inside tolerance loop, so cost has to be at least `4n^3` due to those.

Reply:
Ah, ok. Now I understand where the confusion is arising from. Matrix multiplication has been completely removed from `evd_classic` and there was some wrong commit that reverted those changes in that file. I have reinstated the new version in this pull request. Check it out!!